pkg/mcproducer/scripts/replace-kernel-module.sh

```diff
@@ -19,17 +21,36 @@ EOF
     cat "$kmm_config_file_filepath"
 }
 
-echo "before checking image tar file presence"
-if [ -e $kernel_module_image_filepath ]; then
-    echo "Image file $kernel_module_image_filepath found on the local file system, creating kmm config file"
+echo "before checking image presence"
+if [ -n "$(podman images -q $full_kernel_module_image 2> /dev/null)" ]; then
+    echo "Image $full_kernel_module_image found on the local file system, creating kmm config file"
     create_kmm_config
-    echo "running kernel-management worker image"
-    podman run --user=root --privileged -v /lib/modules:/lib/modules -v $kmm_config_file_filepath:/etc/kmm-worker/config.yaml -v $kernel_module_image_filepath:$kernel_module_image_filepath $worker_image kmod load --tarball /etc/kmm-worker/config.yaml
+    mkdir -p /tmp/opt/lib/modules
+    podman pod create --name kmm-pod
+    echo "creating init container"
+    podman create \
+      --pod kmm-pod \
+      --init-ctr=always \
+      -v /tmp/opt/lib/modules:/tmp/opt/lib/modules:Z \
+      $full_kernel_module_image \
+      /bin/sh -c "cp -R /opt/lib/modules/* /tmp/opt/lib/modules"
+    echo "creating worker container"
+    podman create \
```

Comment:
worker container should also mount the /lib/modules host directory

Reply:
are you sure? even without this mount I was able to install the kmod on my node

Reply:
yes. It is needed in case the container kernel driver is dependent on the host kernel drivers. In your test case, simple-kmod is not dependent on anything, so it is working